compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2529,6 +2592,7 @@ fn add_upstream_rust_crates<'a>(
         let dst = tmpdir.join(cratepath.file_name().unwrap());
         let name = cratepath.file_name().unwrap().to_str().unwrap();
         let name = &name[3..name.len() - 5]; // chop off lib/.rlib
+        let staticlib_suffix = String::from(sess.target.staticlib_suffix.borrow() as &str);
```

Comment:
```suggestion
        let staticlib_suffix = sess.target.staticlib_suffix.clone();
```
(And `f.ends_with(staticlib_suffix.as_ref())` below.)